protocol/x/perpetuals/keeper/perpetual.go

```diff
@@ -29,6 +31,33 @@ import (
 	gometrics "github.com/hashicorp/go-metrics"
 )
 
+// GetInsuranceFundName returns the name of the insurance fund account for a given perpetual.
+// For isolated markets, the name is "insurance-fund:<perpetualId>".
+// For cross markets, the name is "insurance-fund".
+func (k Keeper) GetInsuranceFundName(ctx sdk.Context, perpetualId uint32) (string, error) {
+	perpetual, err := k.GetPerpetual(ctx, perpetualId)
+	if err != nil {
+		return "", err
+	}
+
+	if perpetual.Params.MarketType == types.PerpetualMarketType_PERPETUAL_MARKET_TYPE_ISOLATED {
+		return types.InsuranceFundName + ":" + lib.UintToString(perpetualId), nil
+	} else if perpetual.Params.MarketType == types.PerpetualMarketType_PERPETUAL_MARKET_TYPE_CROSS {
+		return types.InsuranceFundName, nil
+	}
```

Comment:
Question, is our cosmos-sdk version up to date enough for this [code snippet](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-028-public-key-addresses.md#module-account-addresses) to work?

Reply:
Are you referring to the address.Module API?
It should be considering we already use authtypes.NewModuleAddress which uses that underneath

Reply:
I don't think the 2 solutions would be very different in terms of impact (having separate addresses for these accounts) and the latter would require a bigger refactor